### PR TITLE
add support to fmt-v10

### PR DESCRIPTION
**Pull request by daizhirui**
## Type

-   [ ] Bug fix (non-breaking change which fixes an issue): Fixes #
-   [ ] New feature (non-breaking change which adds functionality). Resolves #
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) Resolves #

## Checklist:

-   [x] I have run `python util/check_style.py --apply` to apply Open3D **code style**
    to my code.
-   [ ] This PR changes Open3D behavior or adds new functionality.
    -   [ ] Both C++ (Doxygen) and Python (Sphinx / Google style) **documentation** is
        updated accordingly.
    -   [ ] I have added or updated C++ and / or Python **unit tests** OR included **test
        results** (e.g. screenshots or numbers) here.
-   [x] I will follow up and update the code if CI fails.
    
-   [x] For fork PRs, I have selected **Allow edits from maintainers**.

## Description

Make the build work with fmt-10.2.0 such that "USE_SYSTEM_FMT" works again.

### CODE REVIEW

**Comment by ssheorey** [comment]
Hi @daizhirui can you check the Windows errors?

**Comment by daizhirui** [comment]
@ssheorey I fix several issues and test it with my Windows laptop which uses `CUDA 11.8`:
1. `v10.2.1` is not compatible with CUDA on Windows: use `v10.1.1` instead when `BUILD_CUDA_MODULE` is `ON` on `Windows`.
2. add a missing formatter for `cusolverStatus_t`.

**Comment by daizhirui** [comment]
There are still errors with MSVC 19.29.30154.0, which is not tested on my windows laptop. I also notice there are warnings like
```
MASM : warning A4018: invalid command-line option : /GS- [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\daizh 
\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Zc:wchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\User 
s\daizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Z:wchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users 
\daizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Zwchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\ 
daizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Zchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\d 
aizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Zhar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\da 
izh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Zar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\dai 
zh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
MASM : warning A4018: invalid command-line option : /Zr_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\daiz 
h\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
```
when I test the build with MSVC 14.39.33519 provided by Visual Studio 2022.

I will try to fix the build with Visual Studio 2019. However, is there a plan to update the test environment? Since some compile options configured by CMake are not supported by Visual Studio 2019, I think it is time to upgrade the test environmenet.

**Comment by daizhirui** [comment]
It is the issue of the combination of `NVCC` and `MSVC v142`. I fix it by switching fmt to `6.0.0` when `MSVC v142` is used and `BUILD_CUDA_MODULE` is `ON`.

File: 3rdparty/find_dependencies.cmake [hunk 1]
```diff
@@ -1554,7 +1554,9 @@ if(OPEN3D_USE_ONEAPI_PACKAGES)
         TARGETS TBB::tbb
     )
     list(APPEND Open3D_3RDPARTY_PRIVATE_TARGETS_FROM_SYSTEM Open3D::3rdparty_tbb)
-
+    target_compile_definitions(3rdparty_tbb INTERFACE OPEN3D_USE_ONEAPI_PACKAGES=1)
+    target_compile_definitions(3rdparty_tbb INTERFACE _PSTL_UDR_PRESENT=0)
+    target_compile_definitions(3rdparty_tbb INTERFACE _PSTL_UDS_PRESENT=0)
```
Comment (ssheorey) [inline 3rdparty/find_dependencies.cmake:1559]: Hi @daizhirui what do these compile definitions do and why are they needed?

**Comment by ssheorey** [comment]
> There are still errors with MSVC 14.29.30154.0, which is not tested on my windows laptop. I also notice there are warnings like
> 
> ```
> MASM : warning A4018: invalid command-line option : /GS- [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\daizh 
> \repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Zc:wchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\User 
> s\daizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Z:wchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users 
> \daizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Zwchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\ 
> daizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Zchar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\d 
> aizh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Zhar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\da 
> izh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Zar_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\dai 
> zh\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> MASM : warning A4018: invalid command-line option : /Zr_t [C:\Users\daizh\repos\Open3D\build_on_off_on_release\tbb\src\ext_tbb-build\tbb_static.vcxproj] [C:\Users\daiz 
> h\repos\Open3D\build_on_off_on_release\ext_tbb.vcxproj]
> ```
> 
> which does not appear when I test the build with MSVC 14.39.33519 provided by Visual Studio 2022.
> 
> I will try to fix the build with Visual Studio 2019. However, is there a plan to update the test environment? Since some compile options configured by CMake are not supported by Visual Studio 2019, I think it is time to upgrade the test environmenet.

Thanks for fixing these. We will update MSVC pretty soon.

File: 3rdparty/find_dependencies.cmake (hunk 1 above)
Comment (ssheorey) [reply 3rdparty/find_dependencies.cmake:1559]: This discussion looks relevant, but doesn't talk about what they do:

https://github.com/oneapi-src/oneDPL/issues/1602

Comment (ssheorey) [reply 3rdparty/find_dependencies.cmake:1559]: Ok for now - planning to remove pstl in favor of tbb shortly.

**Review by ssheorey — APPROVED** [review]
Thanks @daizhirui Looks good!

**Pull request merged**